File: backend/src/main/java/com/ums/ums_backend/entity/Course.java
package com.ums.ums_backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

/**
 * Represents a course offered by a department with associated enrollments and instructors.
 */
@Entity
@Getter
@Setter
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;

    @Column(nullable = false, unique = true, length = 10)
    private String code;

    @Column(nullable = false, length = 30)
    private String name;

    @Column(length = 250)
    private String description;

    @Column(nullable = false)
    private Double credits;

    @OneToMany(mappedBy = "course")
    private List<Enrollment> Enrollments = new ArrayList<>();

    @OneToMany(mappedBy = "course")
    private List<CourseInstructor> courseInstructors = new ArrayList<>();

}

File: frontend/src/app/features/course/models/course.ts
import {Department} from '../../department/models/department';
import {Enrollment} from '../../enrollment/models/enrollment';
import {CourseInstructor} from '../../courseInstructor/models/course-instructor';

export interface Course {

  id: number,
  department: Department,
  code: string,
  name: string,
  description: string,
  credits: number,
  enrollments: Enrollment[]
  courseInstructors: CourseInstructor[]

}

export interface CourseCreateRequest {

  departmentId: number,
  code: string,
  name: string,
  description: string,
  credits: number

}

